Sts1CobcSw/Hal/Communication.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <rodos_no_using_namespace.h>
+
+#include <etl/string.h>
+
+#include <concepts>
+#include <cstddef>
+#include <cstdint>
+#include <span>
+#include <string_view>
+#include <type_traits>
+
+
+namespace sts1cobcsw::hal
+{
+template<typename T>
+concept Writable = requires(T t, void const * sendBuf, std::size_t len)
+{
+    // TODO: Check why clang-format fucks this up
+    {
+        t.write(sendBuf, len)
+        } -> std::integral;
+};
+
+
+template<typename T>
+concept ReadWritable =
+    requires(T t, const void * sendBuf, std::size_t len, void * recBuf, std::size_t maxLen)
+{
+    {
+        t.writeRead(sendBuf, len, recBuf, maxLen)
+        } -> std::integral;
+};
```

Comment:
Those concepts are actually not really necessary. I just wrote them because I wanted to play around with concepts. The only thing they do is hopefully give a bit more useful error messages when someone wants to call a function like `WriteTo()` with a type that does not support the correct `write()`. But even without these two concepts the error message should make it quite obvious that you want to `WriteTo()` something that does not support `write()`.